src/main/java/step3/strategy/MoveStrategy.java

```diff
@@ -0,0 +1,5 @@
+package step3.strategy;
+
+public interface MoveStrategy {
```

Comment:
```suggestion
@FunctionalInterface
public interface MoveStrategy {
```

소소하지만 람다로 사용할 수 있다는 한 줄 정도 표시는 붙여줘도 괜찮을 것 같네요.